doc/dev_guide/internals/box_protocol.rst

```diff
@@ -1281,16 +1282,20 @@ each call to conn:new_stream() assigns a new number, starting with 1.
 ..  _box_protocol-stream_transactions:
 
 The client makes stream transactions by sending, in order:
-IPROTO_BEGIN, the transaction data-change and query requests,
-IPROTO_COMMIT or IPROTO_ROLLBACK.
-Each request must contain the same IPROTO_STREAM_ID value.
-With streaming there is no need to add
-:ref:`IPROTO_FLAGS <box_protocol-flags>` and IPROTO_FLAG_COMMIT
-in the header of the last request of a transaction.
-Rollback will be automatic if disconnect occurs before commit is possible.
+
+1. IPROTO_BEGIN with an optional transaction timeout in the IPROTO_TIMEOUT field of the request body
+2. the transaction data-change and query requests
+3. IPROTO_COMMIT or IPROTO_ROLLBACK
+
+All these requests must contain the same IPROTO_STREAM_ID value.
+
+With streaming, there is no need to add
+:ref:`IPROTO_FLAGS <box_protocol-flags>` and IPROTO_FLAG_COMMIT in the header
```

Comment:
I understand that you just rearrange the text written before you, but `IPROTO_FLAG_COMMIT` has **absolutely nothing** to do with client transactions so saying that "there is no need to add `IPROTO_FLAG_COMMIT`" sounds confusing - like you may set it, but it's not necessary. In fact, those flags must not be used in client transactions at all - they will be silently ignored. `IPROTO_FLAG_COMMIT` is an internal flags that is used to store transactions in WAL and replicate them.

Reply:
Got it, removing this sentence.